spec/ics-24-host-requirements/README.md

```diff
@@ -0,0 +1,103 @@
+---
+ics: 24
+title: Host State Machine Requirements
+stage: draft
+category: ibc-core
+requires: 2
+required-by: 2, 3, 4, 5, 18
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2019-04-16
+modified: 2019-04-29
+---
+
+# Synopsis
+
+This specification defines the minimal set of interfaces which must be provided and properties which must be fulfilled by a blockchain & state machine hosting an IBC handler (implementation of the interblockchain communication protocol; see [the architecture document](../../docs/ibc/1_IBC_ARCHITECTURE.md) for details).
+
+# Specification
+
+## Motivation
+
+IBC is designed to be a common standard which will be hosted by a variety of blockchains & state machines and must clearly define the requirements of the host.
+
+## Definitions
+
+`RootOfTrust` is as defined in [ICS 2](../ics-2-consensus-requirements).
+
+## Desired Properties
+
+IBC should require as simple an interface from the underlying state machine as possible to maximize the ease of correct implementation.
+
+## Technical Specification
+
+### Keys, Identifiers, Separators
+
+A `Key` is a bytestring used as the key for an object stored in state. Keys MUST contain only alphanumeric characters and the separator `/`.
+
+An `Identifier` is a bytestring used as a key for an object stored in state, such as a connection, channel, or light client. Identifiers MUST consist of alphanumeric characters only.
+
+Identifiers are not intended to be valuable resources — to prevent name squatting, minimum length requirements or pseudorandom generation MAY be implemented.
```

Comment:
```suggestion
Identifiers are not intended to be valuable resources — to prevent name squatting, minimum length requirements or pseudorandom generation **may** be implemented.
```

Reply:
We want to use the capitalized words - ref https://www.ietf.org/rfc/rfc2119.txt